modules/gridmap/grid_map.cpp

```diff
@@ -561,10 +561,12 @@ bool GridMap::_octant_update(const OctantKey &p_key) {
 				NavigationServer::get_singleton()->region_set_navigation_layers(region, navigation_layers);
 				NavigationServer::get_singleton()->region_set_navmesh(region, navmesh);
 				NavigationServer::get_singleton()->region_set_transform(region, get_global_transform() * nm.xform);
-				if (navigation) {
-					NavigationServer::get_singleton()->region_set_map(region, navigation->get_rid());
-				} else {
-					NavigationServer::get_singleton()->region_set_map(region, get_world()->get_navigation_map());
+				if (is_inside_tree()) {
```

Comment:
Do you need to protect all of this, or just the `NavigationServer::get_singleton()->region_set_map(region, get_world()->get_navigation_map());` call below?

Reply:
All to be more consistent with Godot 4 code and don't add a trap for the next time a line is added.